File: task_two.js
let str= "13 DUP 4 POP 5 DUP + DUP + -"

//let str = '3 DUP 5 - -';

function solution(str) {
  let strArr = str.split(' ');
  let stack = [];

  for (let i = 0; i < strArr.length; i++) {
    if (strArr[i] === 'DUP') {
      // DUPLICATE PREVIOUS IN STACK
      if (emptyArray(stack)) {
        return -1;
      } else {
        let last = stack[stack.length - 1];
        stack.push(last);
      }
    } else if (strArr[i] === 'POP') {
      if (emptyArray(stack)) {
        return -1;
      } else {
        stack.pop(stack[stack.length - 1]);
      }
      // pop the last  element in the stack
    } else if (strArr[i] === '+') {
      //check stack if single element
      if (emptyArray(stack) || checkSingleElement(stack)) {
        return -1;
      } else {
        // add
        let sum = stack[stack.length - 1] + stack[stack.length - 2];
        // slice
        stack.splice(stack.length - 2, 2);
        // push array
        stack.push(sum);
      }
      // add current and previous
    } else if (strArr[i] === '-') {
      //check stack if single element
      // current-previous
      if (emptyArray(stack) || checkSingleElement(stack)) {
        return -1;
      } else {
        // add
        let sub = stack[stack.length - 1] - stack[stack.length - 2];
        if (sub < 0) {
          return -1;
        }
        // slice
        stack.splice(stack.length - 2, 2);
        // push array
        stack.push(sub);
      }
    } else {
      if (checkifInteger(strArr[i])) {
        stack.push(parseInt(strArr[i]));
      } else {
        return -1;
      }
    }
  }

  return emptyArray(stack) ? -1 : stack[stack.length - 1];
}

// check empty
function emptyArray(arr) {
  return arr.length > 0 ? false : true;
}

//check single lement
function checkSingleElement(arr) {
  return arr.length === 1 ? true : false;
}

function checkifInteger(i) {
  return (
    Number.isInteger(parseInt(i)) &&
    parseInt(i) >= 0 &&
    parseInt(i) < Math.pow(2, 20)
  );
}

console.log(solution(str));

File: task_two.py

str= "13 DUP 4 POP 5 DUP + DUP + -"
#str = "5 6 + -"
#str = "3 DUP 5 - -"
# Main function
def solution(str):
    arr = str.split(" ")
    stack = []
    for i in arr:
        #Duplicate element in the stack
        if i == "DUP":
            if empty_stack(stack):
                return -1
            else:
                last = stack[-1]
                stack.append(last)
        elif i == "POP":
            #Remove the last element in the stack
            if empty_stack(stack):
                return -1
            else:
                stack.pop()
        elif i == "+":
            if empty_stack(stack) or check_one_num(stack):
                return -1
            else:
                # subtract the last two numbers in the stack
                sum = stack[-1]+stack[-2]
                # remove the last two numbers in the stack
                del stack[len(stack)-2:len(stack)+1]
                # push the sum to the stack
                stack.append(sum)
        elif i == "-":
            if empty_stack(stack) or check_one_num(stack):
                return -1
            else:
                #subtract the last two numbers in the stack
                sub = stack[-1] - stack[-2]
                # remove the last two numbers in the stack
                del stack[len(stack)-2:len(stack)+1]
                # push the difference to the stack
                stack.append(sub)
        else:
            #add numbers to the stack
            if check_integer(i):
                stack.append(int(i))
    result = -1 if empty_stack(stack) else stack[-1]
    return result

# Helper functions
# helper function to check if stack is empty
def empty_stack(stack):
    result = False if len(stack) > 0  else True
    return result

#helper function to check if the stack has one element
def check_one_num(stack):
    result = True if len(stack) == 1 else False
    return result

# check if the numbers are in the range of 0 to 2^20
def check_integer(i):
    result = True if isinstance(int(i), int) and 0 <= int(i) < 2 ** 20 else False
    return result

#calling our function
print(solution(str))